e2e/nomostest/nt.go

```diff
@@ -139,6 +139,12 @@ type NT struct {
 	// The key is the namespace and name of the RepoSync object, the value points to the corresponding Repository object.
 	NonRootRepos map[types.NamespacedName]*gitproviders.Repository
 
+	// RootSyncHelmCharts is a map of RootSyncs that sync Helm charts
+	RootSyncHelmCharts map[types.NamespacedName]registryproviders.HelmChartID
```

Comment:
With the new `RootSyncHelmCharts`, should we remove `RootRepos` to `RootSyncGitRepos`?

What about OCI repos? Do we want to store them separately?

Reply:
What's the purpose of separating Helm charts from other repos? From this PR, it simplifies the validation a bit. What are other purposes?

My concern is that if we end up having 6 variables for Git/OCI/Helm * 2  (RootSync and RepoSync), it may be simpler to just pass the `rootSha1Fn` and `syncDirectoryMap` to the validation.

Reply:
The old validation method (passing rootSha1Fn and syncDirectoryMap to WatchForAllSyncs) only worked when RootRepos/NonRootRepos contained a record of the RootSync/RepoSync. Otherwise, the RSync would be skipped by WatchForAllSyncs.

I found this hard to read and it made refactoring difficult. So I added a dedicated way to ensure WatchForAllSyncs would validate RSyncs using Helm sources.

I agree that having a single map for all sources would be superior. I just didn't want to also refactor all the OCI tests too in the same PR. That could be a follow up simplification.